components/scream/src/control/tests/sc_tests.cpp

```diff
@@ -0,0 +1,127 @@
+#include "share/grid/point_grid.hpp"
+#include "control/surface_coupling.hpp"
+
+#include <ekat/util/ekat_test_utils.hpp>
+
+#include <catch2/catch.hpp>
+
+TEST_CASE ("surface_coupling")
+{
+  // Some namespaces/aliases
+  using namespace scream;
+  using namespace ShortFieldTagsNames;
+  using namespace ekat::units;
+  using FL = FieldLayout;
+  using FID = FieldIdentifier;
+  using RPDF = std::uniform_real_distribution<Real>;
+  using rngAlg = std::mt19937_64;
+
+  // Some constants
+  constexpr int ncols = 4;
+  constexpr int nlevs = 8;
+  constexpr int nruns = 10;
+
+  // Create a comm
+  ekat::Comm comm (MPI_COMM_WORLD);
+
+  // The random numbers generator
+  std::random_device rd;
+  const unsigned int catchRngSeed = Catch::rngSeed();
+  const unsigned int seed = catchRngSeed==0 ? rd() : catchRngSeed;
+  if (comm.am_i_root()) {
+    std::cout << "seed: " << seed << (catchRngSeed==0 ? " (catch rng seed was 0)\n" : "\n");
+  }
+  rngAlg engine(seed);
+  RPDF pdf(0.0,1.0);
+
+  // Create a grid
+  auto grid = create_point_grid("my grid",ncols*comm.size(), nlevs, comm);
+
+  // Create some field ids, and register them in a field repo
+  // Note: we create two repos, so we can compare outputs with inputs
+  FID s2d_id("s2d",FL{{COL},{ncols}},Pa,grid->name());
+  FID s3d_id("s3d",FL{{COL,VL},{ncols,nlevs}},Pa,grid->name());
+  FID v3d_id("v3d",FL{{COL,CMP,VL},{ncols,2,nlevs}},m/s,grid->name());
+
+  FieldRepository<Real> repo_in, repo_out;
+  repo_in.registration_begins();
+  repo_in.register_field(s2d_id);
+  repo_in.register_field(s3d_id);
+  repo_in.register_field(v3d_id);
+  repo_in.registration_ends();
+  repo_out.registration_begins();
+  repo_out.register_field(s2d_id);
+  repo_out.register_field(s3d_id);
+  repo_out.register_field(v3d_id);
+  repo_out.registration_ends();
+
+  // Create two SC objects, to import and export
+  control::SurfaceCoupling importer(grid,repo_in);
+  control::SurfaceCoupling exporter(grid,repo_out);
+
+  importer.set_num_fields(4,0); // Recall that SC counts *scalar* fields, so vector3d counts as 2 fields
```

Comment:
Could we create an `int` variable that records the count given here as `4`, that way any casual reader of the test is a) able to see the 4 must match the number registered above (+1 for the vector field) and b) is not related to the `ncols` value even farther up.

We could then use the `int` variable instead of hardcoded 4 here.  It also has the bonus that if the test is adjusted in the future to add another field we don't have to track down the hard-coded values too.

Reply:
Sure, I can do this. Note, however, that one will still have to update the variable if the test is updated. But yeah, it is more likely to be remembered that way.